StarVMC/Geometry/FstmGeo/FstmGeoS.xml

```diff
@@ -0,0 +1,1089 @@
+            FHOB,   <!-- Box for FST heat sinks (inside outer wedge) -->
+            FHOC,   <!-- Box for FST heat sinks (inside outer wedge) -->
+            FHOD,   <!-- Box for FST heat sinks (inside outer wedge) -->
+            FHBE,   <!-- Box for FST heat sinks (inside inner wedge tube) -->
+            FHBF,   <!-- Box for FST heat sinks (inside inner wedge tube) -->
+            FHBG,   <!-- Box for FST heat sinks (inside inner wedge tube) -->
+            FHBH,   <!-- Box for FST heat sinks (inside inner wedge tube) -->
+            
+            FSCT,   <!-- Tube for the FST cooling tube -->
+            FSCL,   <!-- Tube for the FST coolant -->
+
+            FSAC,   <!-- Box for FST APV chips -->
+
+            FSTB,   <!-- Box for FST T-board -->
+            FTHC,   <!-- Box for FST T-board connectors -->
+
+            FTIH,   <!-- Tube segment for FST outer hybrid -->
+            FTOH,   <!-- Tube segment for FST inner hybrid -->
+            FSBH,   <!-- Box for FST hybrid -->
+                         
+            FTUS    <!-- Tube segment for FST silicon -->
+        </Content>
+
+        <!-- ////////////////////////////////////////////////////////////////////////////
+   Declare varlists
+        -->
+        <varlist type="int">
+            iwedge/0/, irib/0/, itube/0/, ichip/0/
+        </varlist>
+	
+	<varlist type="float" comment="position of mothervolume and other general offsets">
+	  zcenter/0/ , oapvz, obase2, ibase1, ibase2, iapvz, ibaseAB, zcool
+	</varlist>
+
+	<varlist type="float" comment="offset for cooling tubes ">
+	  dcool /0/
+	</varlist>
+
+
+	<varlist type="float" comment="supportRing Half thickness">
+	sRingThickness/0.3969/ </varlist>  	<!-- Was 0.4  this is 5/16" /2 -->
+
+	<varlist type="float" comment="outer base thickness">
+	oBaseThickness/0.20/ </varlist>  	<!-- Was 0.15 -->
+	<varlist type="float" comment="inner base thickness">
+	iBaseThickness/0.20/ </varlist> 	<!-- Was 0.16 -->
+	<varlist type="float" comment="inner base AB thickness">
+	iBaseThicknessAB/0.06/ </varlist> 	<!-- Was 0.1 -->
+ 	<varlist type="float" comment="peek rib height"> 	<!-- Was 0.25 -->
+	  RibHeight/0.20/
+	</varlist>  
+
+	<!-- these number correspond to drawings as 11-1-2021 -->
+	<varlist type="float" comment="outer base z1">
+	oBaseZ1/14.9442/ </varlist>
+	<varlist type="float" comment="outer base z2">
+	oBaseZ2/30.7605/ </varlist>
+
+	<varlist type="float" comment="HybridThickness"> 
+	HybridThickness/0.035/ </varlist> 	<!-- Was 0.1 -->
+
+	<varlist type="float" comment="Connecting tubes length between inner and outer">
+	  ioTubeLength/0.835/ 	<!-- Has to keep spacing -->
+	</varlist>
+ 
+    <varlist type="float">codeversion/1.1/ </varlist>
+
+
+
+        <!-- ////////////////////////////////////////////////////////////////////////////
+                Declare structures
+        -->
+
+
+        <Structure name="mothervol" comment="tubes for the mother volume">
```

Comment:
In general, structure naming conventions follow the naming convention for volumes.  Four letters, case insensitive.  This is due to  FORtran memory management constraints (aka ZEBRA).  So... 

If you only need this geometry for reconstruction, no action is required.

If you think you would ever run simulations with this geometry, then minimally each structure should start with four unique characters.

Reply:
Are you sure that these data structures need to be 4 characters? I am pretty sure this geometry works in starsim, since that what was used in the last 1.5 year by me for checking it out. In the FstmGeo.F the structure variable do show up with the full name length.